lib/fallback.ts

```diff
@@ -10,7 +10,74 @@ export type FallbackOptions = {
  * Allows to provide a substitute value when method call fails or rejects.
  * @param options optional fallback options
  */
-export function fallback(valueProvider: (...args: any[]) => any, options?: FallbackOptions);
-export function fallback(value: any, options?: FallbackOptions) {
-  throw new Error('Not implemented.');
+export function fallback(valueProvider: (...args: any[]) => any, options?: FallbackOptions): any;
+export function fallback(value: any, options?: FallbackOptions): any;
+export function fallback(value: any | ((...args: any[]) => any), options?: FallbackOptions): any {
+
+  return function (target: any, propertyKey: any, descriptor: PropertyDescriptor) {
+    const method = descriptor.value;
+
+    descriptor.value = function () {
+      try {
+        const result = method.call(this, arguments);
+        const isPromiseLike = result instanceof Promise;
```

Comment:
The recommended way to check if a value is a Promise-like is to check if it has the `.then` function. Current implementation might work only with built-in promise library.